bug: v7 - React - onClick doesn't work in ionButton inside ionSelect

### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

The new functionality insert ionButton inside ionSelect work noramly to visualize but when is insert a function like the onClick dosen't work, the event isn't fired, have othen events that work normaly like onTouch(Start or End).

### Expected Behavior

Just when insert the onClick and click the event be fire.

### Steps to Reproduce

```
<IonSelect
        label='Level 1'
        placeholder="Select Level 1"
        labelPlacement="floating"
        fill="solid"
        value={'test'}
        onIonChange={e=>{}}
>
       <IonSelectOption value="test">Opt</IonSelectOption>
       <IonButton fill="clear" slot="end" onClick={e=>{alert('fire')}}>
          <IonIcon slot="icon-only" color='dark' icon={addCircle}></IonIcon>
       </IonButton>
</IonSelect>
```

### Code Reproduction URL

_No response_

### Ionic Info

[WARN] Error loading @capacitor/ios package.json: Error: Cannot find module '@capacitor/ios/package.json'

       Require stack:
       - C:\Users\Arcano\AppData\Roaming\npm\node_modules\@ionic\cli\lib\project\index.js
       - C:\Users\Arcano\AppData\Roaming\npm\node_modules\@ionic\cli\lib\index.js
       - C:\Users\Arcano\AppData\Roaming\npm\node_modules\@ionic\cli\index.js
       - C:\Users\Arcano\AppData\Roaming\npm\node_modules\@ionic\cli\bin\ionic

Ionic:

   Ionic CLI       : 7.1.1 (C:\Users\Arcano\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework : @ionic/react 7.6.4

Capacitor:

   Capacitor CLI      : 5.4.1
   @capacitor/android : 5.4.1
   @capacitor/core    : 5.4.1
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.3

System:

   NodeJS : v18.16.0 (C:\Program Files\nodejs\node.exe)
   npm    : 9.5.1
   OS     : Windows 10

### Additional Information

_No response_

--- a/core/src/components/select/select.tsx
+++ b/core/src/components/select/select.tsx
@@ -772,12 +772,26 @@ export class Select implements ComponentInterface {
       this.open(ev);
     } else {
       /**
-       * Prevent clicks to the start/end slots from opening the select.
-       * We ensure the target isn't this element in case the select is slotted
-       * in, for example, an item. This would prevent the select from ever
-       * being opened since the element itself has slot="start"/"end".
+       * Clicking a slotted element also causes a click
+       * on the <label> element (since it wraps the slots).
+       * Clicking <label> dispatches another click event on
+       * the native form control that then bubbles up to this
+       * listener. This additional event targets the host
+       * element, so the select overlay is opened.
+       *
+       * When the slotted elements are clicked (and therefore
+       * the ancestor <label> element) we want to prevent the label
+       * from dispatching another click event.
+       *
+       * Do not call stopPropagation() because this will cause
+       * click handlers on the slotted elements to never fire in React.
+       * When developers do onClick in React a native "click" listener
+       * is added on the root element, not the slotted element. When that
+       * native click listener fires, React then dispatches the synthetic
+       * click event on the slotted element. However, if stopPropagation
+       * is called then the native click event will never bubble up
+       * to the root element.
        */
-      ev.stopPropagation();
       ev.preventDefault();
     }
   };

--- a/core/src/components/select/select.tsx
+++ b/core/src/components/select/select.tsx
@@ -772,6 +772,11 @@ export class Select implements ComponentInterface {
       this.open(ev);
     } else {
       /**
+       * Prevent clicks to the start/end slots from opening the select.
+       * We ensure the target isn't this element in case the select is slotted
+       * in, for example, an item. This would prevent the select from ever
+       * being opened since the element itself has slot="start"/"end".
+       *
        * Clicking a slotted element also causes a click
        * on the <label> element (since it wraps the slots).
        * Clicking <label> dispatches another click event on

--- a/core/src/components/select/test/select.spec.tsx
+++ b/core/src/components/select/test/select.spec.tsx
@@ -69,3 +69,31 @@ describe('ion-select', () => {
     expect(slotEl).toBe(null);
   });
 });
+
+describe('select: slot interactivity', () => {
+  test('should not prevent click handlers from firing', async () => {
+    // https://github.com/ionic-team/ionic-framework/issues/28818
+    const divSpy = jest.fn();
+    const buttonSpy = jest.fn();
+
+    const page = await newSpecPage({
+      components: [Select],
+      template: () => (
+        <div onClick={divSpy}>
+          <ion-select label="Label Prop Text">
+            <button slot="end" onClick={buttonSpy}>
+              Button
+            </button>
+          </ion-select>
+        </div>
+      ),
+    });
+
+    const button = page.body.querySelector('button')!;
+
+    await button.click();
+
+    expect(buttonSpy).toHaveBeenCalled();
+    expect(divSpy).toHaveBeenCalled();
+  });
+});

